src/python/pants/backend/project_info/peek.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import json
+import os
+from dataclasses import asdict, is_dataclass
+from enum import Enum
+from itertools import repeat
+from typing import Iterable, Mapping, Tuple, TypeVar, Union, cast
+
+from pkg_resources import Requirement
+
+from pants.engine.addresses import Address, Addresses, BuildFileAddress
+from pants.engine.console import Console
+from pants.engine.fs import DigestContents, FileContent, PathGlobs
+from pants.engine.goal import Goal, GoalSubsystem
+from pants.engine.rules import Get, MultiGet, collect_rules, goal_rule
+from pants.engine.target import Target, Targets
+
+
+class OutputOptions(Enum):
+    RAW = "raw"
+    JSON = "json"
+
+
+class PeekSubsystem(GoalSubsystem):
```

Comment:
We'll want to use the `Outputting` mixin so that the `--output-file` option is added: https://www.pantsbuild.org/v2.2/docs/rules-api-goal-rules#outputting-mixin-optional.

Reply:
Okay, seems simple enough

- [ ] add `Outputting` mixin